contributors/design-proposals/kubeadm-upgrades.md

```diff
@@ -0,0 +1,350 @@
+   - **Optional policy**: The kubeadm CLI patch version should be equal to or higher than the upgrade target patch version.
+     - Example: kubeadm CLI version `v1.8.3` can upgrade a cluster to `v1.8.3` without an explicit grant, but upgrading to `v1.8.5` requires forcing. 
+ - The kubeadm CLI of version `vX.Y.Z` can create clusters with minor versions `Y` and `Y-1`.
+   - This is a **required policy**, and is not specific to upgrades. This policy is also is also enforced when creating a cluster.
+     - Example: kubeadm CLI version `v1.8.2` can create clusters of (and upgrade clusters to) version `v1.8.x` and `v1.7.x`.
+ - The control plane version must always be higher than or equal to the version of all the kubelets, and the maximum allowed skew is one minor versions.
+   - **Required policy**: The control plane version must be higher than or equal to the kubelet version. This is a generic Kubernetes policy.
+     - Example: For a control plane of version `v1.8.4`; a kubelet version may be `v1.8.2`, but not `v1.9.3`
+   - **Skippable policy**: The maximum allowed minor skew between the control plane and the kubelets is one minor version.
+     - Example: For a control plane of version `v1.7.2` being upgraded to `v1.8.5`; the minimum allowed kubelet version is `v1.7.0`.
+     - Note: We might relax this policy to allow two minor versions before going to GA.
+ - The version change must be positive, i.e. only upgrades are supported; not downgrades or runtime reconfiguration.
+   - This is a **skippable policy** in v1.8, and will be removed in future versions before going to GA.
+     - Example: Downgrading from `v1.8.3` to `v1.8.2` or reconfiguring a cluster on version `v1.8.1` can only be done (initially) along with the `--force` flag.
+ - The maximum allowed upgrade minor skew is one minor version.
+   - This is a **required policy**. You can upgrade from `vX.Y.Z1` to `vX.Y+1.Z2`, but not to `vX.Y+2.Z3`.
+     - Example: Upgrading from `v1.6.2` to `v1.8.4` is not supported.
+     - Note: We might relax this policy to allow two minor versions before going to GA.
+ - The target upgrade version must not be an alpha, beta version unless the user explicitely allows it with `--allow-experimental-upgrades`
+   - This is a **skippable policy**, in order to have an extra safety check but still allow it for users who may want it. Such upgrades are not supported.
+     - Example: Upgrading from `v1.7.1` to `v1.8.0-beta.3` can only be done with at least one of the `--allow-experimental-upgrades` or `--force` flags.
+ - The target upgrade version must not be a release candidate version unless the user explicitely allows it with `--allow-release-candidate-upgrades` (or `--allow-experimental-upgrades`)
+   - This is a **skippable policy**, in order to have an extra safety check but still allow users to easily test our RCs. These upgrades "should work".
```

Comment:
does "should work" mean that we do support these (as opposed to upgrades to alpha / beta releases)?

Reply:
Well I mean that the risk should be low, but we can't support these cases (alpha is alpha, right?)

Reply:
in that case make it explicit (as above) that these upgrades are generally expected to work but are also officially unsupported.